compiler/transforms/acc_cse.py

```diff
@@ -53,15 +54,90 @@ def match_and_rewrite(self, op: acc.SetupOp, rewriter: PatternRewriter, /):
         )
 
 
-class AccCse(ModulePass):
+class HoistSetupCallsIntoConditionals(RewritePattern):
     """
-    Common subexpression elimination for the `acc` (accelerator) dialect.
+    Hoists setup calls into scf.if blocks.
```

Comment:
```suggestion
    Hoists setup calls into scf.if blocks, preceding the setup calls.
```
(it took a while before I realized this)